backend/src/oe/api/op-blockspan.service.ts

```diff
@@ -0,0 +1,140 @@
+    try {
+      const confirmedBlocks = [] as ConfirmedBlockHeight[];
+      const currentTip = await bitcoinApi.$getBlockHeightTip();
+
+      blockSpanList.forEach((blockNumber) =>
+        confirmedBlocks.push(
+          opBlockHeaderService.verifyConfirmedBlockHeight(
+            blockNumber.startBlockHeight,
+            {
+              value: currentTip,
+            }
+          ),
+          opBlockHeaderService.verifyConfirmedBlockHeight(
+            blockNumber.endBlockHeight,
+            {
+              value: currentTip,
+            }
+          )
+        )
+      );
+      return confirmedBlocks;
+    } catch (e) {
+      throw new Error(`${UUID} OpBlockSpanApiService.$getConfirmedBlockSpanList: error while filtering confirmed block span list: ${e instanceof Error ? e.message : e}`);
+    }
+  }
+
+  public async $generateBlockSpanDetailedList(UUID: string, endBlockHeight: number, span: number, blockSpanCount: number): Promise<BlockSpanDetails[]> {
+    try {
+      const blockSpanList = this.$getBlockSpanList(
+        UUID,
+        endBlockHeight,
+        span,
+        blockSpanCount
+      );
+      const confirmedBlockSpanList = await this.$getConfirmedBlockSpanList(UUID, blockSpanList);
+      if (!confirmedBlockSpanList.length) {
+        return [];
+      }
+
+      const blockHeadersList = await opBlockHeaderService.$getBlockHeadersByHeights(
+        UUID,
+        Array.from(new Set(confirmedBlockSpanList))
+      );
+
+      let index = 0;
+      const result: BlockSpanDetails[] = [];
+      while (index < blockHeadersList.length - 1) {
+        result.push({
+          startBlock: {
+            height: blockHeadersList[index].height,
+            hash: blockHeadersList[index].current_block_hash,
+            mediantime: blockHeadersList[index].mediantime,
+            timestamp: blockHeadersList[index].timestamp,
+          },
+          endBlock: {
+            height: blockHeadersList[++index].height,
+            hash: blockHeadersList[index].current_block_hash,
+            mediantime: blockHeadersList[index].mediantime,
+            timestamp: blockHeadersList[index].timestamp,
+          },
+        });
+      }
+      return result;
+    } catch (e) {
+      throw new Error(`${UUID} OpBlockSpanApiService.$generateBlockSpanDetailedList: error while generating detailed block span list: ${e instanceof Error ? e.message : e}`);
+    }
+  }
+
+  public async $getBlockSpanDetailedList(
+    UUID: string,
+    endBlockHeight: number,
+    span: number,
+    blockSpanCount: number,
+    withNbdrStatistics: boolean
+  ): Promise<BlockSpanDetails[] | BlockSpanDetailsWithNbdr[] | NbdrStatisticsError> {
+    try {
+      const blockSpanDetailedList = await this.$generateBlockSpanDetailedList(
+        UUID,
+        endBlockHeight,
+        span,
+        withNbdrStatistics && blockSpanCount !== -1 ? blockSpanCount + 100 : blockSpanCount
```

Comment:
I suggest not using 0 in any case. Else in the future, it will create confusion for us also. Even if anyhow user enters 0 we will be sure that we have handled that scenario. So let's keep -1 for all blocks and 0 as not allowed